src/test/fastfilter_tests.cpp

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2018 The Bitcoin Unlimited developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "test/test_bitcoin.h"
+
+#include <boost/test/unit_test.hpp>
+
+#include "fastfilter.h"
+#include "hash.h"
+
+BOOST_FIXTURE_TEST_SUITE(fastfilter_tests, BasicTestingSetup)
+
+BOOST_AUTO_TEST_CASE(fastfilter_tests)
+{
+    // Like a bloom filter the fast filter can have false positives but not false negatives
+    FastRandomContext insecure_rand;
+    {
+        CFastFilter<1024 * 1024> filt;
+
+        //  pick a random start point for a randomized test
+        // arith_uint256 num(insecure_rand.rand32());
+        arith_uint256 num(1);
+        arith_uint256 origNum = num;
+        int collisions = 0;
+        for (int i = 1; i < 20000; i++)
+        {
+            num += 1;
+            uint256 t1 = ArithToUint256(num);
+            // for the fastfilter to work without lots of collisions the data must be pseudo-random
+            uint256 tmp = Hash(t1.begin(), t1.end());
+            if (filt.contains(tmp))
+            {
+                collisions++;
+            }
+            filt.insert(tmp);
+            BOOST_CHECK(filt.contains(tmp));
+            BOOST_CHECK(!filt.checkAndSet(tmp));
+        }
+        // printf("collisions: %d\n", collisions);
+        BOOST_CHECK(collisions < 400); // sanity check, actual result may vary
```

Comment:
could remove commented printf...there are a couple more down below

Reply:
I'm going to remove them for expediency but we need to have a conversation about this.  I think that test code should be directed towards 2 uses:
1. regression test
2. figuring out what went wrong

In the context of 2, leaving some helpful statements commented out makes a lot of sense.

Reply:
But didn't the test print out the values of `collision` in case of failure since we set `log_level` default value is `error`. 

https://www.boost.org/doc/libs/1_54_0/libs/test/doc/html/utf/user-guide/runtime-config/reference.html